inc/templates/admin-bar.php

```diff
@@ -348,3 +349,50 @@ function get_documentation_items(): array {
 		],
 	];
 }
+
+/**
+ * Get an array of Irving cache busting node args to be passed into
+ * admin_bar->add_node().
+ *
+ * @return array
+ */
+function get_cache_items(): array {
+	return [
+		[
+			'id'    => 'irving-cache-purge',
+			'title' => __( 'Clear The Cache', 'wp-irving' ),
+			'href'  => '#',
+		],
+	];
+}
+
+/**
+ * Register the click listener for the `irving-cache-purge` node. Register
+ * an action event, `wp_ajax_irving_cache_purge`, on click using WP ajax.
+ */
+function cache_purge_click_listener() {
+	?>
+		<script type="text/javascript">
+			jQuery('#wp-admin-bar-irving-cache-purge').on('click', function() {
+				const data = { 'action': 'irving_cache_purge' };
+				jQuery.post( '<?php echo esc_url( admin_url( 'admin-ajax.php' ) ); ?>', data );
+			});
+		</script>
+	<?php
+}
+add_action( 'admin_footer', __NAMESPACE__ . '\cache_purge_click_listener', 95 );
+
+/**
+ * Purge the cache. Called when the `wp_ajax_irving_cache_purge` action
+ * is fired.
+ */
+function cache_purge_callback() {
```

Comment:
Re: above, this logic should move into an endpoint, and it'd be great to split it up into "clear everything" & "clear this".

Reply:
Interesting thought. I'll dig into that!